web/src/typing.ts

```diff
@@ -71,13 +92,33 @@ function send_typing_notification_based_on_message_type(
     to: Recipient,
     operation: "start" | "stop",
 ): void {
+    if (to.notification_event_type !== "typing") {
+        return;
+    }
```

Comment:
Should this just be an `assert to.notification_event_type === "typing"`? 

Writing this logic with a `return` statement makes one have to think about whether returning is correct that case (and what that case is), so if it's intended to be impossible, we should just say that.

Reply:
thanks for letting me know! updated it.